[inductor][easy] create a wrap for triton do_bench function

shunting314 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #99385
* #97203
* __->__ #99216

triton PR https://github.com/openai/triton/pull/1513 change the interface of do_bench function. The quantile fields name is changed from 'percentiles' to 'quantiles' and its default value is changed from from (0.5, 0.2, 0.8) to None. This break some inductor code since a caller expects a tuple may get a item.

Add a wrapper to maintain the same behavior for inductor.






cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

shunting314 committed:
[inductor][easy] create a wrap for triton do_bench function

[ghstack-poisoned]

jansel reviewed (changes requested):
Let's just change our wrapper to return a single value rather than a tuple.

I believe all our usage just ignores the extra values.

shunting314 committed:
Update on "[inductor][easy] create a wrap for triton do_bench function"


triton PR https://github.com/openai/triton/pull/1513 change the interface of do_bench function. The quantile fields name is changed from 'percentiles' to 'quantiles' and its default value is changed from from (0.5, 0.2, 0.8) to None. This break some inductor code since a caller expects a tuple may get a item.

Add a wrapper to maintain the same behavior for inductor.






[ghstack-poisoned]

shunting314 commented:
I did a `grep -r 'bench('` under `torch/_inductor` directory and update all the call sites to assume a item returned from do_bench.

I also checked `torch/_dynamo`, `benchmarks/dynamo` folder, looks like `torch._inductor.testing.do_bench` is not used there.

shunting314 committed:
Update on "[inductor][easy] create a wrap for triton do_bench function"


triton PR https://github.com/openai/triton/pull/1513 change the interface of do_bench function. The quantile fields name is changed from 'percentiles' to 'quantiles' and its default value is changed from from (0.5, 0.2, 0.8) to None. This break some inductor code since a caller expects a tuple may get a item.

Add a wrapper to maintain the same behavior for inductor.






cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

shunting314 commented:
@pytorchbot merge -f "SEGFAULT and model accuracy check fail are unrelated"
